### PR TITLE
remove code intel from secondary locations to ease extension API removal

### PR DESCRIPTION
**Pull request by sqs**
This removes code intel from marginal places to make it easier to fully remove the legacy extension API support.

Until we re-add it, these places will no longer have code intel (hovers):

- Batch Changes diff snippets
- Notebooks snippets
- Diffs (commits, comparisons)

Right now, these are implemented in a way that depends on the legacy extensions controller API. We want to remove that fully. That migration will go much, much faster if we don't need to maintain the legacy implementation for these secondary and lesser-used places.

See [more internal context](https://sourcegraph.slack.com/archives/C03P1V3J92P/p1671673841946569).

## Test plan

View diffs in all of the places mentioned above. Confirm that they work but do not show code intel.

## App preview:

- [Web](https://sg-web-sqs-rm-extension-api-secondary-uis.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| -1.47% (-41.99 kb) 🔽 | -0.92% (-136.70 kb) 🔽 | -0.79% (-94.71 kb) 🔽 | -1.05% (-8) 🔽 |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/sqs/rm-extension-api-secondary-uis/compare-report.html;tab=live_object) for a full comparison between the commits 9999374 and 39703ad or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/repo/compare/RepositoryCompareArea.tsx [hunk 1]
```diff
@@ -91,61 +71,11 @@ export interface RepositoryCompareAreaPageProps extends PlatformContextProps {
...
-    ): RepoSpec & RevisionSpec & ResolvedRevisionSpec & FileSpec & UIPositionSpec & ModeSpec {
-        return {
-            repoName: hoveredToken.repoName,
-            revision: hoveredToken.revision,
-            filePath: hoveredToken.filePath,
-            commitID: hoveredToken.commitID,
-            position: hoveredToken,
-            mode: getModeFromPath(hoveredToken.filePath || ''),
-        }
+        this.state = {}
```
[the remark this answers is not in the document]
Comment (valerybugakov) [reply client/web/src/repo/compare/RepositoryCompareArea.tsx:78]: I asked a model to refactor this component to the functional component and remove component's state:

```tsx
export const RepositoryCompareArea: React.FunctionComponent<RepositoryCompareAreaProps> = ({
    repo,
    setBreadcrumb,
    match,
    location,
    history,
    platformContext,
    isLightTheme,
}: RepositoryCompareAreaProps) => {
    useBreadcrumb(useMemo(() => ({ key: 'compare', element: <>Compare</> }), []))

    let spec: { base: string | null; head: string | null } | null | undefined
    if (match.params.spec) {
        spec = parseComparisonSpec(decodeURIComponent(match.params.spec))
    }
    const searchParams = new URLSearchParams(location.search)
    const path = searchParams.get('filePath')

    if (!repo) {
        return <LoadingSpinner />
    }

    const commonProps: RepositoryCompareAreaPageProps = {
        repo,
        base: { repoID: repo.id, repoName: repo.name, revision: spec?.base },
        head: { repoID: repo.id, repoName: repo.name, revision: spec?.head },
        routePrefix: match.url,
        platformContext,
    }

    return (
        <div className={classNames('container', styles.repositoryCompareArea)}>
            <RepositoryCompareHeader className="my-3" {...commonProps} />
            {spec === null ? (
                <Alert variant="danger">Invalid comparison specifier</Alert>
            ) : (
                <Switch>
                    <Route
                        path={`${match.url}`}
                        key="hardcoded-key" // see https://github.com/ReactTraining/react-router/issues/4578#issuecomment-334489490
                        exact={true}
                        render={routeComponentProps => (
                            <RepositoryCompareOverviewPage
                                {...routeComponentProps}
                                {...commonProps}
                                path={path}
                                isLightTheme={isLightTheme}
                            />
                        )}
                    />
                    <Route key="hardcoded-key" component={NotFoundPage} />
                </Switch>
            )}
        </div>
    )
}
```

**Review by valerybugakov — APPROVED** [review]
Awesome cleanup! Thank you for doing that.

File: client/web/src/components/diff/Lines.tsx [hunk 2]
```diff
@@ -69,11 +64,8 @@ export const Line: React.FunctionComponent<React.PropsWithChildren<Line>> = ({
     id,
     anchor,
     html,
-    lineStyle,
-    decorations,
     className,
     dataPart,
-    isLightTheme,
```
[the remark this answers is not in the document]
Comment (sqs) [reply client/web/src/components/diff/Lines.tsx:76]: Done. Found ~200 other deletions I could make. Nice!

**Comment by sqs** [comment]
Thank you for the review, @valerybugakov! You pointed out about ~350 extra lines that could be deleted safely.

**Comment by philipp-spiess** [comment]
Wooohohoo

**Pull request merged**